src/vintch_model/subunit_model.py

```diff
@@ -0,0 +1,178 @@
+from typing import Any
+from .backend_config import get_backend
+from .nonlinearity import TentNonlinearity
+from einops import einsum
+
+# type_ArrayLike = Union[np.ndarray, "jax.numpy.ndarray", "torch.Tensor"]
+
+
+class SubunitModel:
+    """
+    Implementation of the Vintch model.
+
+    Parameters
+    ----------
+    kernel_size :
+        Size of the kernel filters. Defaults to 8.
+    kernel_time_depth :
+        Time depth of the kernel filters. Defaults to 8.
+    n_dim :
+        Number of dimension of the input, defaults to 16. This is also the dimension
+        of the pooling weights.
+    n_basis_funcs :
+        Number of basis functions for the nonlinearity. Defaults to 25.
+    backend :
+        Computational backend to use, either 'jax' or 'torch'. Defaults to 'torch'.
+    n_channels :
+        Number of channels in the model. Defaults to 2.
+    is_channel_excitatory :
+        List indicating whether each channel is excitatory. Defaults to [True, False].
+    pooling_in_time :
+        Whether to perform weight pooling also in time. Defaults to True.
+    """
+
+    def __init__(
+        self,
+        kernel_size: int = 8,
+        kernel_time_depth: int = 8,
+        n_dim: int = 16,
+        n_basis_funcs: int = 25,
+        backend: str = "torch",
+        n_channels: int = 2,
+        is_channel_excitatory: list = [True, False],
+        pooling_in_time: bool = True,
+        kernels: Any = None,
+        pooling_weights: Any = None,
+        pooling_biases: Any = None,
+    ):
+        assert (
+            len(is_channel_excitatory) == n_channels
+        ), "Length of is_channel_excitatory must match n_channels"
+        self.kernel_size = kernel_size
+        self.kernel_time_depth = kernel_time_depth
+        self.n_dim = n_dim
+        self.n_channels = n_channels
+        self.backend_class = get_backend(backend)
+        self._kernels = self.backend_class.randn(
+            (n_channels, 1, kernel_time_depth, kernel_size, kernel_size)
+        )
+        if pooling_in_time:
+            self._pooling_weights = self.backend_class.randn(
+                (n_channels, n_dim, n_dim, n_dim)
```

Comment:
I think this shape should be constructed similarly to what I said about kernel size about: have a single `pooling_size` argument, which is a tuple of ints of arbitrary shape. that also allows us to get rid of the pooling_in_time argument: if the length of this input is the same as `kernel_size`, then we're pooling in time. If not, it must be `len(kernel_size)-1` and we're not pooling in time